sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -56,11 +56,10 @@ def _convert_datetime_columns_to_numeric(self, data, metadata):
                         data[column_name] = pd.to_datetime(
                             data[column_name], format=datetime_format
                         )
-
-                    data.loc[~pd.isna(data[column_name]), column_name] = pd.to_numeric(
-                        data[column_name]
-                    )
-                    data[column_name] = data[column_name].fillna(np.nan)
+                    nan_mask = pd.isna(data[column_name])
+                    data[column_name] = pd.to_numeric(data[column_name])
+                    if nan_mask.any():
```

Comment:
I'm only running this if there are nans because it will change the dtype from int to float if there aren't